File: src/main/java/com/meicloud/flink/stream/utils/CommonStreamUtils.java
package com.meicloud.flink.stream.utils;


import com.meicloud.spark.entity.CaseVo.*;
import com.meicloud.spark.utils.ConstantUtils;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.api.java.tuple.Tuple5;

import java.util.HashMap;

public class CommonStreamUtils {


    public static String getCheckpointLocation(JobConfigVo jobConfigVo) {
        return new StringBuilder()
                .append(ConstantUtils.CHECKPOINT_LOCATION())
                .append("_")
                .append(jobConfigVo.jobName())
                .append("_")
                .append(jobConfigVo.version()).toString();
    }

    public static void validateParameter(Tuple5<JobConfigVo, Tuple2<String, InputSourceConfigVo>, Tuple2<String, ExecutorConfigVo>, HashMap<String, OutputSourceConfigVo>, HashMap<String, StaticInputSourceConfigVo>> jobAllConfigTuple) {
        if (jobAllConfigTuple.f0 == null) {
            System.err.println("Usage: 没有配置JOB");
            System.exit(1);
        }
        if (jobAllConfigTuple.f1 == null) {
            System.err.println("Usage: 没有配置输入源");
            System.exit(1);
        }
        if (jobAllConfigTuple.f2 == null) {
            System.err.println("Usage: 没有配置执行层");
            System.exit(1);
        }
        if (jobAllConfigTuple.f3 == null || jobAllConfigTuple.f3.size() == 0) {
            System.err.println("Usage: 没有此配置输出源");
            System.exit(1);
        }
    }


}

File: src/main/scala/com/meicloud/spark/stream/util/CommonStreamUtils.scala
package com.meicloud.spark.stream.util

import java.util.UUID

import com.meicloud.spark.utils.{ConstantUtils, _}
import com.meicloud.spark.entity.CaseVo._
import org.apache.spark.sql.SparkSession

import scala.collection.mutable

/**
  * Created by yesk on 2019-3-10. 
  */
object CommonStreamUtils {

  def validateParameter(jobAllConfigTuple: (JobConfigVo, (String, InputSourceConfigVo),
    (String, ExecutorConfigVo), mutable.HashMap[String, OutputSourceConfigVo], mutable.HashMap[String, StaticInputSourceConfigVo])): Unit = {
    if (jobAllConfigTuple._1 == null) {
      System.err.println("Usage: 没有配置JOB")
      System.exit(1)
    }
    if (jobAllConfigTuple._2 == null) {
      System.err.println("Usage: 没有配置输入源")
      System.exit(1)
    }
    if (jobAllConfigTuple._3 == null) {
      System.err.println("Usage: 没有配置执行层")
      System.exit(1)
    }
    if (jobAllConfigTuple._4 == null || jobAllConfigTuple._4.size == 0) {
      System.err.println("Usage: 没有此配置输出源")
      System.exit(1)
    }
  }

  def getCheckpointLocation(jobConfigVo: JobConfigVo): String = {
    new StringBuffer().append(PropertiesScalaUtils.getString(ConstantUtils.CHECKPOINT_LOCATION))
      .append(jobConfigVo.jobName)
      .append("_")
      .append(jobConfigVo.version.toString).toString
  }

  def getSparkSession(jobConfigVo: JobConfigVo): SparkSession = {
    val spark = SparkSession
      .builder
      //      .master(if (jobConfigVo.master==null || "".equals(jobConfigVo.master)) "yarn" else jobConfigVo.master)
      .master("local")
      .appName(jobConfigVo.jobName)
      //设置shuffle的个数

      .config("spark.yarn.tags", jobConfigVo.jobName)
      .getOrCreate()
    spark.sparkContext.setLogLevel(ConstantUtils.LOG_LEVEL_INFO)
    //    spark.sqlContext.setConf("spark.sql.shuffle.partitions","100")
    spark
  }
}
